<launch>
   <param name="/use_sim_time" value="false" />
   <node name="turtle_driver" pkg="turtle_driver" type="turtle_driver" output="screen" respawn="true">
      <rosparam file="$(find turtle_driver)/cfg/my_turtle_params_imu.yaml" command="load" />
   </node>

   <include file="$(find turtle_driver)/launch/robot_pose_ekf.launch"/>

  <arg name="node_name"             value="velocity_smoother"/>
  <arg name="nodelet_manager_name"  value="nodelet_manager"/>
  <arg name="config_file"           value="$(find turtle_driver)/cfg/yocs_velocity_smoother.yaml"/>
  <arg name="raw_cmd_vel_topic"     value="cmd_vel"/>
  <arg name="smooth_cmd_vel_topic"  value="smoother_cmd_vel"/>
  <arg name="robot_cmd_vel_topic"   value="robot_cmd_vel"/>
  <arg name="odom_topic"            value="odom"/>
  
  <!-- nodelet manager -->
  <node pkg="nodelet" type="nodelet" name="$(arg nodelet_manager_name)" args="manager"/>
  
  <!-- velocity smoother -->
  <include file="$(find yocs_velocity_smoother)/launch/velocity_smoother.launch">
    <arg name="node_name"             value="$(arg node_name)"/>
    <arg name="nodelet_manager_name"  value="$(arg nodelet_manager_name)"/>
    <arg name="config_file"           value="$(arg config_file)"/>
    <arg name="raw_cmd_vel_topic"     value="$(arg raw_cmd_vel_topic)"/>
    <arg name="smooth_cmd_vel_topic"  value="$(arg smooth_cmd_vel_topic)"/>
    <arg name="robot_cmd_vel_topic"   value="$(arg robot_cmd_vel_topic)"/>
    <arg name="odom_topic"            value="$(arg odom_topic)"/>
  </include>

  <node pkg="tf" type="static_transform_publisher" name="imu_broadcaster" args="0 0 0 0 0 0 base_footprint imu_base 100" /> 
</launch>
